app/scripts/controllers/exportImport.js

```diff
@@ -28,29 +29,59 @@ angular.module('openhimConsoleApp')
       $scope.showRecordOptions.Mediators = false;
       $scope.showRecordOptions.ContactGroups = false;
     };
+    
+    var getMetadataSuccess = function(result) {
+      exportObject = result[0];
+      var Users = exportObject.Users;
+      var Clients = exportObject.Clients;
+      var Channels = exportObject.Channels;
+      var Mediators = exportObject.Mediators;
+      var ContactGroups = exportObject.ContactGroups;
+
+      // set up settings object
+      $scope.exportSettings = {};
+      $scope.exportSettings.removeIds = true;
+
+      // Assign API collections to export options object
+      $scope.exportCollections = {};
+      $scope.exportCollections.Users = Users;
+      $scope.exportCollections.Clients = Clients;
+      $scope.exportCollections.Channels = Channels;
+      $scope.exportCollections.Mediators = Mediators;
+      $scope.exportCollections.ContactGroups = ContactGroups;
+
+      // set inital reset ( default export option )
+      $scope.resetExportOptions();
+    };
+    
+    var getMetadataError = function(err) {
+      Alerting.AlertReset();
+      Alerting.AlertAddMsg('top', 'danger', 'An error has occurred while fetching metadata: #' + err.status + ' - ' + err.data);
+    };
 
+    var openValidationModal = function() {
+      var modalInstance = $modal.open({
+        templateUrl: 'views/exportImportModal.html',
+        controller: 'ExportImportModalCtrl',
+        resolve: {
+          data: function () {return $scope.validatedData;}
+        }
+      });
+
+      modalInstance.result.then(function(importResults) {
+        $scope.importStatus = 'done';
+        $scope.importResults = importResults;
+      }, function() {
+        console.info('Modal dismissed at: ' + new Date());
+      });
+    };      
+      
     // Make API requests for the export configuration options
-    var Users = Api.Users.query();
-    var Clients = Api.Clients.query();
-    var Channels = Api.Channels.query();
-    var Mediators = Api.Mediators.query();
-    var ContactGroups = Api.ContactGroups.query();
-
-    // set up settings object
-    $scope.exportSettings = {};
-    $scope.exportSettings.removeIds = true;
-
-    // Assign API collections to export options object
-    $scope.exportCollections = {};
-    $scope.exportCollections.Users = Users;
-    $scope.exportCollections.Clients = Clients;
-    $scope.exportCollections.Channels = Channels;
-    $scope.exportCollections.Mediators = Mediators;
-    $scope.exportCollections.ContactGroups = ContactGroups;
-
-    // set inital reset ( default export option )
-    $scope.resetExportOptions();
-    
+    var exportObject = Api.Metadata.query(function(result) {
+      getMetadataSuccess(result);
+    }, function(err) {
+      getMetadataError(err);
+    });
```

Comment:
Could you not simply do `var exportObject = Api.Metadata.query(getMetadataSuccess, getMetadataError)`?